plutus-use-cases/test/Spec/Crowdfunding.hs

```diff
@@ -192,11 +181,14 @@ startingBalance = 1000
 checkCFTrace :: CFScenario -> Trace EmulatedWalletApi () -> Property
 checkCFTrace CFScenario{cfInitialBalances} t = property $ do
     let model = Gen.generatorModel { Gen.gmInitialBalance = cfInitialBalances }
-    (result, st) <- forAll $ Gen.runTraceOn model t
+    (result, st) <- forAll $ Gen.runTraceOn model (processPending >>= notifyBlock >> t)
     Hedgehog.assert (isRight result)
     Hedgehog.assert ([] == emTxPool st)
 
--- | Validate all pending transactions and notify the wallets
-updateAll :: CFScenario -> Trace EmulatedWalletApi [Tx]
-updateAll CFScenario{cfWallets} =
-    processPending >>= walletsNotifyBlock cfWallets
+-- | Notify all wallets in the campaign of the new blocks.
+notifyBlocks :: [Block] -> Trace EmulatedWalletApi ()
+notifyBlocks = traverse_ notifyBlock
+
+-- | Notify all wallets in the campaign of a new block
+notifyBlock :: Block -> Trace EmulatedWalletApi ()
+notifyBlock = void . walletsNotifyBlock [w1, w2, w3]
```

Comment:
I wonder if it would simplify our tests to add a `getWallets` action that you can run only in `EmulatedWalletApi`? Then you obviously can't use it from things with a `WallletApi` constraint only, but you could put things in the test that would just work generically over all the wallets (`EmulatedWalletApi` already knows about them because they've got state inside it).

Reply:
Yeah I think I'll do that (new PR)

Reply:
Yeah, no need for it to be in here.

Reply:
https://iohk.myjetbrains.com/youtrack/issue/CGP-426